src/compiler/types.ts

```diff
@@ -1861,6 +1861,53 @@ namespace ts {
         getJsxIntrinsicTagNames(): Symbol[];
         isOptionalParameter(node: ParameterDeclaration): boolean;
 
+        /**
+         * Two types are identical iff they are references to the exact same fully qualified, resolved type (usually an alias to ===)

```

Comment:
Most often, the identity relation will only be true when `a === b`. There's only a few other things (identical generic instantiations) not covered by that.